src/test/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/PerformingOrganisationNameAndAddressTest.java

```diff
@@ -0,0 +1,77 @@
+package uk.nhs.digital.nhsconnect.lab.results.model.edifact;
+
+import org.junit.jupiter.api.Test;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.message.EdifactValidationException;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+class PerformingOrganisationNameAndAddressTest {
+
+    private final PerformingOrganisationNameAndAddress performingOrganisationNameAndAddress =
+        new PerformingOrganisationNameAndAddress("LONDON CITY HOSPITAL");
+
+    @Test
+    void when_edifactStringDoesNotStartWithCorrectKey_expect_illegalArgumentExceptionIsThrown() {
+        assertThrows(IllegalArgumentException.class, () -> PerformingOrganisationNameAndAddress.fromString("wrong value"));
+    }
+
+    @Test
+    void when_edifactStringIsPassed_expect_returnAPerformingOrganisationNameAndAddressObject() {
+        assertThat(performingOrganisationNameAndAddress)
+            .usingRecursiveComparison()
+            .isEqualTo(PerformingOrganisationNameAndAddress.fromString("NAD+SLA+++LONDON CITY HOSPITAL"));
+    }
+
+    @Test
+    void when_mappingSegmentObjectToEdifactString_expect_returnCorrectEdifactString() {
+        String expectedEdifactString = "NAD+SLA+++LONDON CITY HOSPITAL'";
+
+        PerformingOrganisationNameAndAddress performingOrganisation = PerformingOrganisationNameAndAddress.builder()
+                .performingOrganisationName("LONDON CITY HOSPITAL")
+                .build();
+
+        assertEquals(expectedEdifactString, performingOrganisation.toEdifact());
+    }
+
+    @Test
+    void when_mappingSegmentObjectToEdifactStringWithEmptyField_expect_edifactValidationExceptionIsThrown() {
+        PerformingOrganisationNameAndAddress performingOrganisation = PerformingOrganisationNameAndAddress.builder()
+                .performingOrganisationName("")
+                .build();
+
+        assertThrows(EdifactValidationException.class, performingOrganisation::toEdifact);
```

Comment:
I know we're doing it to our custom exception, but should we also check the exception message to make sure that the exception is being called where we expect it to be called? 

I'm not against this way of testing the exception though - more a point of curiosity.

Reply:
This is testing the `toEdifact` method in the `Segment` superclass. We verify the exception message in the `testPreValidate` test below